### PR TITLE
Add ability to switch amoung multiple course sets

### LINKED ISSUE #27: transcript import broken on safari
.text() doesn't work on files in safari

### LINKED ISSUE #31: Changing theme on the schedule page does not change the calendar's font colors

### PR DESCRIPTION
**Pull request by elihschiff**
also fixes #27 and fixes  #31

### CODE REVIEW

File: src/store/modules/schedule.ts [hunk 1]
```diff
@@ -28,9 +32,70 @@ export default class Schedule extends VuexModule {
...
+      }
+    }
+  }
+
+  @Mutation
+  newCourseSet(p: { name: string }): void {
+    Vue.set(this.courseSets[this.currentTerm], p.name, {});
+  }
+
+  @Action
```
Comment (benjaminrsherman) [inline src/store/modules/schedule.ts:61]: This can be a mutation I think.

File: src/components/CourseSetEdit.vue [hunk 2]
```diff
@@ -0,0 +1,158 @@
+<template>
+  <div>
+    <b-nav-item-dropdown left title="Switch between saved courseSets">
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:3]: Change title to `Switch between saved course sets`.

File: src/components/CourseSetEdit.vue [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+           The page nicely spaced out and not bouncing-->
+        <b-form-valid-feedback id="valid-feedback">
+          Must be a unique name
+        </b-form-valid-feedback>
+      </b-input-group>
+
+      <h3 style="margin: 0px;">Current course sets:</h3>
+      <p style="font-size: 80%;" v-if="Object.keys(getCourseSets).length <= 1">
+        There must always be 1 course set, add another to remove the current
+        courseSet
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue]: Change to `course set`.

File: src/components/CourseSetEdit.vue [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    "b-input-group": BInputGroup,
+    "b-input-group-append": BInputGroupAppend,
+  },
+  directives: {
+    "b-modal": VBModal,
+  },
+  computed: {
+    ...mapGetters("schedule", ["getCourseSets"]),
+    ...mapState("schedule", ["currentCourseSet", "courseSets"]),
+    verifyNewCourseSet(): boolean {
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:120]: This should probably be renamed to something like `newCourseSetExists`.

File: src/store/modules/schedule.ts [hunk 5]
```diff
@@ -28,9 +32,70 @@ export default class Schedule extends VuexModule {
...
+      worker.setSelected(sec, false);
+    }
+    this.currentCourseSet = p.name;
+    for (const sec in this.courseSets[this.currentTerm][
+      this.currentCourseSet
+    ]) {
+      if (this.courseSets[this.currentTerm][this.currentCourseSet][sec]) {
+        worker.setSelected(sec, true);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (elihschiff) [reply src/store/modules/schedule.ts:53]: You cannot call a mutation from a mutation, so I am pretty sure I do the regen conflict stuff from where this function is called instead. I dont really have a great solution to this problem.

File: src/store/modules/schedule.ts (hunk 1 above)
Comment (elihschiff) [reply src/store/modules/schedule.ts:61]: same with what I said above, you cannot call a mutation from a mutation. vuex is dumb

File: src/main.ts [hunk 6]
```diff
@@ -21,8 +21,10 @@ import {
   faCaretRight,
   faCheck,
   faCog,
+  faEdit,
```
[the remark this answers is not in the document]
Comment (elihschiff) [reply src/main.ts:24]: Ok we can do this at some point after this pr

File: src/components/CourseSetEdit.vue [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      >
+    </b-nav-item-dropdown>
+
+    <b-modal id="courseSet-modal" title="Course Set Settings">
+      <p></p>
+      <b-input-group>
+        <b-form-input
+          v-model="newCourseSetName"
+          :state="verifyNewCourseSet"
+          placeholder="CourseSet Name"
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:35]: This should be `Course Set Name`.

File: .github/workflows/lint.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        run: git submodule update --remote --recursive
+
+      - name: Clean install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Install wasm-pack
+        run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+
+      - name: Run linter
+        run: yarn lint
```
Comment (benjaminrsherman) [inline .github/workflows/lint.yml:28]: This should be `yarn build` since that does a more comprehensive check.

**Pull request merged**